### PR TITLE
feat: add steady-state throughput windows

### PR DESCRIPTION
**Pull request by ben1009**
## Summary

Add per-second throughput window summaries to steady-state write-perf JSON output and finish the v2 shape for the remaining steady-state workloads.

## Changes

- Add optional throughput records with operation, read, write, logical-byte, read-byte, write-byte, and scan-row rate percentiles.
- Align worker start timing with a barrier so complete-window rates do not include thread spawn skew.
- Convert point_read_missing_in_range to the steady-state v2 measurement shape with task, latency, throughput, validation, and drain metadata.
- Add an idle steady-state workload for background drain and baseline measurement rows.
- Keep read key selection allocation-free while supporting reserved-padding missing keys.

## Verification

- cargo test --package kv-engine --bin write-perf
- cargo run --package kv-engine --bin write-perf -- --suite steady-state --preset smoke --bench readmissing_in_range --warmup-secs 0 --measurement-secs 1 --latency-sample-every 1000 --output json --path /tmp/toykv-missing-rate-window-smoke-2
- cargo run --package kv-engine --bin write-perf -- --suite steady-state --preset smoke --bench balanced --warmup-secs 0 --measurement-secs 1 --latency-sample-every 1000 --output json --path /tmp/toykv-balanced-rate-window-smoke-2
- git diff --check
- cargo make check

## Summary by CodeRabbit

- **New Features**
  - Added steady-state idle workload measurement.
  - Added throughput reporting for operations, reads, writes, bytes, and scan rows.
  - Added percentile throughput statistics to serialized results.
  - Added missing-key workload support with warmup, closed-loop measurement, and validation metadata.

- **Tests**
  - Added coverage for workload selection, throughput serialization, and percentile calculations.

### CODE REVIEW

**Comment by ben1009** [comment]
@coderabbitai  review

File: kv-engine/src/bin/write-perf.rs [hunk 1]
```diff
@@ -2599,6 +2827,63 @@ fn latency_record(
...
+        ),
+        write_logical_bytes: rate_window_record(
+            stats
+                .rate_windows
+                .iter()
+                .map(|window| window.write_logical_bytes),
+        ),
+        scan_rows: rate_window_record(stats.rate_windows.iter().map(|window| window.scan_rows)),
+    })
+}
```
[the remark this answers is not in the document]
Comment (ben1009) [reply kv-engine/src/bin/write-perf.rs:2862]: Fixed. Throughput windows are now attributed by the operation start bucket instead of the completion time, so work issued during the final measurement second is counted even if it completes after the stop signal. I also added the worker-release helper and focused tests for the window merge/empty-window branches.
